src/benchmarks/micro/libraries/System.IO/StringReaderReadToEndTests.cs

```diff
@@ -0,0 +1,28 @@
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.IO.Tests
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class StringReaderReadToEndTests : TextReaderReadLineTests
+    {
+        [GlobalSetup]
+        public void GlobalSetup() => _text = GenerateLinesText(LineLengthRange, 16 * 1024);
+
+        [Benchmark]
+        public void ReadLine()
+        {
+            using StringReader reader = new (_text);
```

Comment:
Do we want to test the construction of the `StringReader` here too or should it be moved to setup methods?

Reply:
StringReader has a _pos field so we can't really move it up.